packages/amplify-util-uibuilder/src/clients/amplify-studio-client.ts

```diff
@@ -77,7 +77,7 @@ export default class AmplifyStudioClient {
   /**
    * Used to configure the AWS Amplify clients.
    */
-  static async setClientInfo(context: $TSContext, envName?: string, appId?: string): Promise<AmplifyStudioClient> {
+  static async setClientInfo(context: $TSContext, envName?: string, appId?: string): Promise<AmplifyStudioClient & {isSupportedGraphQL?: boolean}> {
```

Comment:
Can the type for `AmplifyStudioClient` be extended? Because you're directly attaching this key value pair to the client object. For this type to be accurate, I'd expect a return below like:

```ts
    let isGraphQLSupported = false;
    if ((await getTransformerVersion()) === 2) {
      await client.loadMetadata();
      isGraphQLSupported = true;
    }
    return { ...client, isGraphQLSupported };
```

Side note, this method should really not be called `setClientInfo` if it both configures and returns a client, but it doesn't need to be addressed in this PR.

Reply:
I added this because of a TS error. Not sure how it wasn't added before.
The question mark makes it optional, which works for the current implementation.

Reply:
Actually I'm able to resolve this another way.